src/Containers.jsx

```diff
@@ -179,9 +179,9 @@ class Containers extends React.Component {
         let emptyCaption = _("No running containers");
         const containersStats = this.props.containersStats;
         // TODO: check filter text
-        let filtered = [];
-        Object.keys(this.props.containers).filter(id => { if (!this.props.onlyShowRunning || this.props.containers[id].State.Running) { filtered.push(this.props.containers[id]) } });
-        let rows = filtered.map((ctr, id) => this.renderRow(containersStats, ctr));
+        let filtered = Object.keys(this.props.containers).filter(id => !this.props.onlyShowRunning || this.props.containers[id].State.Running)
+                .reduce((obj, id) => { return {...obj, [id]: this.props.containers[id]} }, {});
```

Comment:
The first line is totally fine to build a list `filtered` that has the IDs of everything you want to show. You can directly use it in `let rows = filtered.map(id => ...)`. But now this second line converts the `filtered` list into an object (in a *very* expensive way!), and then the `Object.keys()` below turns it back into a list.

Please don't do this kind of black magic. If you have to copy and convert between data structures a lot, then this is a sign that your data structures are wrong. In this case they are actually okay (you keep the state information as a dictionary, and for rendering you filter them into a list), you are just doing some extra work and then more to undo it :)